snowshu/core/replica/replica_factory.py

```diff
@@ -118,10 +119,13 @@ def _execute(self,
             graph_to_result_list(graphs),
             self.run_analyze)
 
-    def load_config(self, config: Union[Path, str, TextIO]):
+    def load_config(self,
+                    config: Union[Path, str, TextIO],
+                    target_arch=None):
         """does all the initial work to make the resulting ReplicaFactory
         object usable."""
         logger.info('Loading configuration...')
         start_timer = time.time()
         self.config = ConfigurationParser().from_file_or_path(config)
+        self.config.target_profile.adapter.shdocker = SnowShuDocker(target_arch)
```

Comment:
There were 3 or 4 separate shdocker inits in target adapter. This was the best place IMO to place one, since adapter's init is in the config parser. Putting it there would be good if the aim was to select architectures in replica.yml, but ticket is for CLI.